postgis/lwgeom_remove_irrelevant_points_for_view.c

```diff
@@ -0,0 +1,289 @@
+/**********************************************************************
+ *
+ * PostGIS - Spatial Types for PostgreSQL
+ * http://postgis.net
+ *
+ * PostGIS is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * PostGIS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PostGIS.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **********************************************************************
+ *
+ * Copyright (C) 2024 Sam Peters <gluser1357@gmx.de>
+ *
+ **********************************************************************/
+
+#include "postgres.h"
+#include "funcapi.h"
+#include "utils/array.h"
+#include "utils/builtins.h"
+#include "utils/lsyscache.h"
+#include "utils/numeric.h"
+#include "access/htup_details.h"
+
+#include "liblwgeom.h"
+#include "liblwgeom_internal.h"
+
+// ===============================================================================
+// helper function for polygon and polyline POINTARRAY's
+// which removes points being irrelevant for rendering the geometry within
+// a view specified by rectangular bounds.
+// ===============================================================================
+void ptarray_remove_helper(POINTARRAY *points, GBOX *bounds, int minpoints);
+void ptarray_remove_helper(POINTARRAY *points, GBOX *bounds, int minpoints) {
```

Comment:
I guess you could define this as static and avoid the declaration

Reply:
Thanks, is fixed now (here and in PR 768).